File: MantenimientoDaxDirectorio/MantenimientoDirectorio/DirectorioAlex.vb
Option Strict Off
Option Explicit On
Imports DattCom
Public Class DirectorioAlex
    'variables locales para almacenar los valores de las propiedades
    Public codigo As String 'copia local
    Public nombres As String 'copia local
    Public apellidos As String 'copia local
    Public NombreImpresion As String 'copia local
    Public CiRuc As String 'copia local
    Public telefono1 As String 'copia local
    Public telefono2 As String 'copia local
    Public direccion As String 'copia local
    Public CodVendedor As String 'copia local
    Public CodCobrador As String 'copia local
    Public ZonaVentas As String 'copia local
    Public ZonaCobranzas As String 'copia local
    Public TipoCliente As String
    Public EsProveedor As Boolean
    Public EsCliente As Boolean
    Public EsEmpleado As Boolean
    Public EsBanco As Boolean
    Public EsVendedor As Boolean
    Public Ciudad As String
    Public PrecioVenta As Short
    Public dESCUENTO As Double
    Public dESCUENTO_prov As Double
    Public FormaPago As String
    Public FormaPago_prov As String
    Public limitecredito As Double
    Public limitecredito_prov As Double
    Public PorcComision As Integer
    Public CtaCobComision As String
    Public TipoId As String
    Public NombreCli As String
    Public TipoPersona As String
    Public ExoneradoIva As Boolean
    Public correoElectronico As String
    Public sector As String

    Public Function CargarAlex(ByRef CodigoAlex As String, ByRef ClioPro As Boolean, Optional ByRef SoloCodigo As String = "") As Boolean
        CargarAlex = False
        Mainn()
        Dim Seleccion As String = ""
        Try
            Dim RsCli = New Identificacion(datosEmpresa.strConxAdcom)
            RsCli = Identificacion.Buscar(" codigo='" & CodigoAlex & "' or cedulaidentidadruc ='" & CodigoAlex & "'  " & Seleccion & "")
            If RsCli Is Nothing Then Exit Function
            Seleccion = ""

            With RsCli
                codigo = .Codigo
                nombres = .Nombres
                apellidos = .Apellidos
                NombreImpresion = .NombreImpresion
                CiRuc = .CedulaIdentidadRuc
                telefono1 = .Telefono1
                telefono2 = .Telefono2
                direccion = .Domicilio
                EsCliente = .EsCliente
                EsProveedor = .EsProveedor
                EsEmpleado = .EsEmpleado
                EsBanco = .EsBanco
                Ciudad = .Ciudad
                EsVendedor = .EsVendedor
                PorcComision = .ComisionVenta
                CtaCobComision = .CtaCobrarComisiones
                TipoId = .TipoIdentificacion
                TipoPersona = .TipoPersona
                ExoneradoIva = .ExoneradoIva
                NombreCli = nombres & " " & apellidos
                correoElectronico = .CorreoElectrónico
                sector = .Sector
            End With
            CargarAlex = True
        Catch exx As Exception
            MsgBox("excepcion en lectura de datos Alex identificacion" + vbCr + exx.Message)
        End Try

        Try
            Dim ConxAdcom As New SqlClient.SqlConnection(datosEmpresa.strConxAdcom)
            ConxAdcom.Open()
            Dim Ssql As String = "SELECT * FROM Cliente WHERE codigocli='" & CodigoAlex & "'"
            Dim comm As New SqlClient.SqlCommand(Ssql, ConxAdcom)
            Dim rs As SqlClient.SqlDataReader = comm.ExecuteReader
            If rs.Read Then
                With rs
                    If Not IsDBNull(.Item("VendedorInterno").ToString) Then CodVendedor = .Item("VendedorInterno").ToString
                    If Not IsDBNull(.Item("TipoCli").ToString) Then TipoCliente = .Item("TipoCli").ToString
                    If Not IsDBNull(.Item("ZonaVentas").ToString) Then ZonaVentas = .Item("ZonaVentas").ToString
                    If Not IsDBNull(.Item("ZonaCobranza").ToString) Then ZonaCobranzas = .Item("ZonaCobranza").ToString
                    If Not IsDBNull(.Item("CobradorInterno").ToString) Then CodCobrador = .Item("CobradorInterno").ToString
                    If Not IsDBNull(.Item("PrecioVenta")) Then PrecioVenta = CorrijeNuloN(.Item("PrecioVenta"))
                    If Not IsDBNull(.Item("PorcDescuento")) Then dESCUENTO = .Item("PorcDescuento")
                    If Not IsDBNull(.Item("FormaPago").ToString) Then FormaPago = .Item("FormaPago").ToString
                    If Not IsDBNull(.Item("limitecredito")) Then limitecredito = CDbl("0" + .Item("limitecredito").ToString)
                End With
            End If
            rs.Close()
        Catch exx As Exception
            MsgBox("excepcion en lectura de datos Alex cliente" + vbCr + exx.Message)
        End Try

        Try
            Dim ConxAdcom As New SqlClient.SqlConnection(datosEmpresa.strConxAdcom)
            ConxAdcom.Open()
            Dim Ssql As String = "SELECT isnull(FormaPago,'') as FormaPago,isnull(PorceDescuent,0) as PorceDescuent,isnull(LimCreditoPrv,0) as LimCreditoPrv FROM proveedor WHERE codigoproveedor='" & CodigoAlex & "'"
            Dim comm As SqlClient.SqlCommand = New SqlClient.SqlCommand(Ssql, ConxAdcom)
            Dim rs As SqlClient.SqlDataReader = comm.ExecuteReader
            If rs.Read Then
                With rs
                    If Not IsDBNull(.Item("PorceDescuent")) Then dESCUENTO_prov = .Item("PorceDescuent")
                    If Not IsDBNull(.Item("FormaPago")) Then FormaPago_prov = .Item("FormaPago").ToString
                    If Not IsDBNull(.Item("LimCreditoPrv")) Then limitecredito_prov = Val(.Item("LimCreditoPrv").ToString)
                End With
            End If
            rs.Close()
        Catch exx As Exception
            MsgBox("excepcion en lectura de datos Alex proveedor" + vbCr + exx.Message)
        End Try

    End Function
End Class

File: MantenimientoDaxDirectorio/MantenimientoDirectorio/DirectorioAlex.cs
using System;
using System.Runtime.InteropServices;
using DattCom;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace directMnt
{
	public class DirectorioAlex
	{
		// variables locales para almacenar los valores de las propiedades
		public string codigo; // copia local
		public string nombres; // copia local
		public string apellidos; // copia local
		public string NombreImpresion; // copia local
		public string CiRuc; // copia local
		public string telefono1; // copia local
		public string telefono2; // copia local
		public string direccion; // copia local
		public string CodVendedor; // copia local
		public string CodCobrador; // copia local
		public string ZonaVentas; // copia local
		public string ZonaCobranzas; // copia local
		public string TipoCliente;
		public bool EsProveedor;
		public bool EsCliente;
		public bool EsEmpleado;
		public bool EsBanco;
		public bool EsVendedor;
		public string Ciudad;
		public short PrecioVenta;
		public double dESCUENTO;
		public double dESCUENTO_prov;
		public string FormaPago;
		public string FormaPago_prov;
		public double limitecredito;
		public double limitecredito_prov;
		public int PorcComision;
		public string CtaCobComision;
		public string TipoId;
		public string NombreCli;
		public string TipoPersona;
		public bool ExoneradoIva;
		public string correoElectronico;
		public string sector;

		public bool CargarAlex(ref string CodigoAlex, ref bool ClioPro, [Optional, DefaultParameterValue("")] ref string SoloCodigo)
		{
			bool CargarAlexRet = default;
			CargarAlexRet = false;
			Module1.Mainn();
			string Seleccion = "";
			try
			{
				var RsCli = new Identificacion(datosEmpresa.strConxAdcom);
				RsCli = Identificacion.Buscar(" codigo='" + CodigoAlex + "' or cedulaidentidadruc ='" + CodigoAlex + "'  " + Seleccion + "");
				if (RsCli is null)
					return CargarAlexRet;
				Seleccion = "";

				codigo = RsCli.Codigo;
				nombres = RsCli.Nombres;
				apellidos = RsCli.Apellidos;
				NombreImpresion = RsCli.NombreImpresion;
				CiRuc = RsCli.CedulaIdentidadRuc;
				telefono1 = RsCli.Telefono1;
				telefono2 = RsCli.Telefono2;
				direccion = RsCli.Domicilio;
				EsCliente = RsCli.EsCliente;
				EsProveedor = RsCli.EsProveedor;
				EsEmpleado = RsCli.EsEmpleado;
				EsBanco = RsCli.EsBanco;
				Ciudad = RsCli.Ciudad;
				EsVendedor = RsCli.EsVendedor;
				PorcComision = (int)Math.Round(RsCli.ComisionVenta);
				CtaCobComision = RsCli.CtaCobrarComisiones;
				TipoId = RsCli.TipoIdentificacion;
				TipoPersona = RsCli.TipoPersona;
				ExoneradoIva = RsCli.ExoneradoIva;
				NombreCli = nombres + " " + apellidos;
				correoElectronico = RsCli.CorreoElectrónico;
				sector = RsCli.Sector;
				CargarAlexRet = true;
			}
			catch (Exception exx)
			{
				Interaction.MsgBox("excepcion en lectura de datos Alex identificacion" + Constants.vbCr + exx.Message);
			}

			try
			{
				var ConxAdcom = new System.Data.SqlClient.SqlConnection(datosEmpresa.strConxAdcom);
				ConxAdcom.Open();
				string Ssql = "SELECT * FROM Cliente WHERE codigocli='" + CodigoAlex + "'";
				var comm = new System.Data.SqlClient.SqlCommand(Ssql, ConxAdcom);
				var rs = comm.ExecuteReader();
				if (rs.Read())
				{
					if (!(rs["VendedorInterno"].ToString() is DBNull))
						CodVendedor = rs["VendedorInterno"].ToString();
					if (!(rs["TipoCli"].ToString() is DBNull))
						TipoCliente = rs["TipoCli"].ToString();
					if (!(rs["ZonaVentas"].ToString() is DBNull))
						ZonaVentas = rs["ZonaVentas"].ToString();
					if (!(rs["ZonaCobranza"].ToString() is DBNull))
						ZonaCobranzas = rs["ZonaCobranza"].ToString();
					if (!(rs["CobradorInterno"].ToString() is DBNull))
						CodCobrador = rs["CobradorInterno"].ToString();
					if (!(rs["PrecioVenta"] is DBNull))
						PrecioVenta = (short)Math.Round(Valores.CorrijeNuloN(rs["PrecioVenta"]));
					if (!(rs["PorcDescuento"] is DBNull))
						dESCUENTO = Conversions.ToDouble(rs["PorcDescuento"]);
					if (!(rs["FormaPago"].ToString() is DBNull))
						FormaPago = rs["FormaPago"].ToString();
					if (!(rs["limitecredito"] is DBNull))
						limitecredito = Conversions.ToDouble("0" + rs["limitecredito"].ToString());
				}
				rs.Close();
			}
			catch (Exception exx)
			{
				Interaction.MsgBox("excepcion en lectura de datos Alex cliente" + Constants.vbCr + exx.Message);
			}

			try
			{
				var ConxAdcom = new System.Data.SqlClient.SqlConnection(datosEmpresa.strConxAdcom);
				ConxAdcom.Open();
				string Ssql = "SELECT isnull(FormaPago,'') as FormaPago,isnull(PorceDescuent,0) as PorceDescuent,isnull(LimCreditoPrv,0) as LimCreditoPrv FROM proveedor WHERE codigoproveedor='" + CodigoAlex + "'";
				var comm = new System.Data.SqlClient.SqlCommand(Ssql, ConxAdcom);
				var rs = comm.ExecuteReader();
				if (rs.Read())
				{
					if (!(rs["PorceDescuent"] is DBNull))
						dESCUENTO_prov = Conversions.ToDouble(rs["PorceDescuent"]);
					if (!(rs["FormaPago"] is DBNull))
						FormaPago_prov = rs["FormaPago"].ToString();
					if (!(rs["LimCreditoPrv"] is DBNull))
						limitecredito_prov = Conversion.Val(rs["LimCreditoPrv"].ToString());
				}
				rs.Close();
			}
			catch (Exception exx)
			{
				Interaction.MsgBox("excepcion en lectura de datos Alex proveedor" + Constants.vbCr + exx.Message);
			}

			return CargarAlexRet;

		}
	}
}